workflow/validate/validate.go

```diff
@@ -484,6 +484,9 @@ func validateInputs(tmpl *wfv1.Template) (map[string]interface{}, error) {
 		if art.From != "" {
 			return nil, errors.Errorf(errors.CodeBadRequest, "templates.%s.%s.from not valid in inputs", tmpl.Name, artRef)
 		}
+		if art.FromExpression != "" {
```

Comment:
good stuff -  I don't think we should put much effort into validation, as long as the workflow fails